hedera-node/src/main/java/com/hedera/services/state/expiry/renewal/ExpiredEntityClassification.java

```diff
@@ -0,0 +1,29 @@
+package com.hedera.services.state.expiry.renewal;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+public enum ExpiredEntityClassification {
+	DETACHED_ACCOUNT,
+	EXPIRED_ACCOUNT_READY_TO_RENEW,
+	DETACHED_ACCOUNT_GRACE_PERIOD_OVER,
+	DETACHED_TREASURY_GRACE_PERIOD_OVER_BEFORE_TOKEN,
```

Comment:
Even if say token `0.0.9999` and its treasury `0.0.1111` had the same expiry, it would still be possible for the auto-renew cycle to find `0.0.1111` first...but it would be a much rarer event than in the current state, where we don't enforce the `treasury.expiry >= token.expiry` relationship.

Reply:
Makes sense, thanks!